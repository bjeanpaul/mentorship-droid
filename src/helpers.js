import { find, isUndefined, partialRight as partial } from 'lodash';
import { getBaseURL } from 'src/configuration';
import { omitBy, isNull } from 'lodash';


export const omitNulls = d => omitBy(d, isNull);


export const trap = (type, fn) => v => v instanceof type
  ? Promise.resolve(v).then(fn)
  : Promise.reject(v);


export const switchError = cases => (e, ...xargs) => {
  const [type, fn] = find(cases, ([type, fn]) => e instanceof type) || [];
  return !isUndefined(fn)
    ? Promise.resolve().then(() => fn(e, ...xargs))
    : Promise.reject(e);
};


export const apiAction = ({
  method,
  request,
  success,
  failures,
}) => (...args) => (dispatch, { auth }) => Promise.resolve()
  .then(() => request(...args))
  .then(dispatch)
  .then(() => method(...args, auth))
  .then(partial(success, ...args), partial(switchError(failures), ...args))
  .then(dispatch);


export const staticAction = type => () => ({
  type
});


export const dataAction = type => data => ({
  type,
  payload: data
});


export const generateActionTypes = function generateActionNames(
  resource,
  operations = ['FETCH', 'CREATE', 'UPDATE', 'DELETE']
) {
  const actionTypes = {};
  const resourceUC = resource.toUpperCase();

  operations.forEach((operation) => {
    const operationUC = operation.toUpperCase();
    const request = `${resourceUC}_${operationUC}_REQUEST`;
    const success = `${resourceUC}_${operationUC}_SUCCESS`;
    const failure = `${resourceUC}_${operationUC}_FAILURE`;

    const operationLC = operation.toLowerCase();
    actionTypes[`${operationLC}Request`] = request;
    actionTypes[`${operationLC}Success`] = success;
    actionTypes[`${operationLC}Failure`] = failure;
  });
  return actionTypes;
};


export const filterActionTypes = function filterActionTypes(
  actionTypes,
  operation
) {
  const [request, success, failure] = [
    `${operation}Request`,
    `${operation}Success`,
    `${operation}Failure`,
  ];

  if (actionTypes[request] || actionTypes[success] || actionTypes[failure]) {
    return [
      actionTypes[request],
      actionTypes[success],
      actionTypes[failure],
    ];
  }
  return [];
};


// TODO: Delete
export const generateActionCreators = function generateActionCreators({
  resourcePath,
  actionTypes,
  requestOpts = {},
  normalizeJSON,
}) {
  const actionCreators = {};
  let url = `${getBaseURL()}/`;
  if (resourcePath) {
    url += `${resourcePath}/`;
  }

  const fetchActionTypes = filterActionTypes(actionTypes, 'fetch');
  if (fetchActionTypes.length === 3) {
    actionCreators.fetch = (onSuccess, onFailure) => ({
      type: '--generated fetch--',
      types: fetchActionTypes,
      url,
      requestOpts,
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }

  const createActionTypes = filterActionTypes(actionTypes, 'create');
  if (createActionTypes.length === 3) {
    actionCreators.create = ({
      body,
      onSuccess,
      onFailure,
    }) => ({
      type: '--generated create--',
      types: createActionTypes,
      url,
      requestOpts: { ...requestOpts,
        method: 'POST',
        body: JSON.stringify(body),
      },
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }

  // id is required as an argument to update a record;
  // we have a scenario with profiless, where it's
  // profile/{id}/image,
  // which is awkward...
  const updateActionTypes = filterActionTypes(actionTypes, 'update');
  if (updateActionTypes.length === 3) {
    actionCreators.update = ({
      id,
      body,
      onSuccess,
      onFailure,
    }) => ({
      type: '--generated update--',
      types: updateActionTypes,
      url: `${url}${id}/`,
      requestOpts: { ...requestOpts,
        method: 'PUT',
        body: JSON.stringify(body),
      },
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }


  return actionCreators;
};
