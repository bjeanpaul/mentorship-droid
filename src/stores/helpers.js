import moment from 'moment';
import { values, isUndefined, sortBy, filter, first } from 'lodash';


export const getAuthUserProfile = ({
  auth: { profileId },
  entities: { profiles },
}) => !isUndefined(profileId)
  ? profiles[profileId]
  : null;


export const getContext = state => {
  const { auth: { auth } } = state;

  return {
    auth,
    profile: getAuthUserProfile(state),
  };
};


// TODO sort?
export const getCategories = ({ entities: { categories } }) => values(categories);


// TODO sort?
export const getCategoryActivities = ({
  entities: { activities },
}, targetCategoryId) => (
  values(activities)
    .filter(({ category }) => category === targetCategoryId));


export const getCategory = ({ entities: { categories } }, id) => categories[id];


export const getActivity = ({ entities: { activities } }, id) => activities[id];


export const getScheduledCalls = ({ entities: { scheduledCalls } }) => values(scheduledCalls);


export const getScheduledCall = ({ entities: { scheduledCalls } }, id) => scheduledCalls[id];


export const getScheduledCallActivity = (state, id) => {
  const scheduledCall = getScheduledCall(state, id);
  return scheduledCall && scheduledCall.activity && getActivity(state, scheduledCall.activity);
};


export const getEvents = ({ entities: { events } }) => values(events);


export const getActivityCallNotes = ({
  entities: { callNotes },
}, activityId) => (
  filter(callNotes, { callActivity: activityId }));


export const getCall = ({ entities: { calls } }, id) => calls[id];


export const getCallNote = ({ entities: { callNotes } }, id) => callNotes[id];


export const getNextScheduledCall = (state, predicate = {}, time = Date.now()) => {
  let res = sortBy(getScheduledCalls(state), ({ callTime }) => callTime);
  res = res.filter(({ callTime }) => moment(callTime).isSameOrAfter(time));
  res = filter(res, predicate);
  return first(res);
};
