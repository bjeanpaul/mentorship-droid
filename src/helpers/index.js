import { getBaseURL } from 'src/configuration';
import { omitBy, isNull } from 'lodash';


export const conj = (...args) => Object.assign({}, ...args);


export const omitNulls = d => omitBy(d, isNull);


export const generateActionTypes = function generateActionNames(
  resource,
  operations = ['FETCH', 'CREATE', 'UPDATE', 'DELETE']
) {
  const actionTypes = {};
  const resourceUC = resource.toUpperCase();

  operations.forEach((operation) => {
    const operationUC = operation.toUpperCase();
    const request = `${resourceUC}_${operationUC}_REQUEST`;
    const success = `${resourceUC}_${operationUC}_SUCCESS`;
    const failure = `${resourceUC}_${operationUC}_FAILURE`;

    const operationLC = operation.toLowerCase();
    actionTypes[`${operationLC}Request`] = request;
    actionTypes[`${operationLC}Success`] = success;
    actionTypes[`${operationLC}Failure`] = failure;
  });
  return actionTypes;
};


export const filterActionTypes = function filterActionTypes(
  actionTypes,
  operation
) {
  const [request, success, failure] = [
    `${operation}Request`,
    `${operation}Success`,
    `${operation}Failure`,
  ];

  if (actionTypes[request] || actionTypes[success] || actionTypes[failure]) {
    return [
      actionTypes[request],
      actionTypes[success],
      actionTypes[failure],
    ];
  }
  return [];
};


// TODO: Delete
export const generateActionCreators = function generateActionCreators({
  resourcePath,
  actionTypes,
  requestOpts = {},
  normalizeJSON,
}) {
  const actionCreators = {};
  let url = `${getBaseURL()}/`;
  if (resourcePath) {
    url += `${resourcePath}/`;
  }

  const fetchActionTypes = filterActionTypes(actionTypes, 'fetch');
  if (fetchActionTypes.length === 3) {
    actionCreators.fetch = (onSuccess, onFailure) => ({
      type: '--generated fetch--',
      types: fetchActionTypes,
      url,
      requestOpts,
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }

  const createActionTypes = filterActionTypes(actionTypes, 'create');
  if (createActionTypes.length === 3) {
    actionCreators.create = ({
      body,
      onSuccess,
      onFailure,
    }) => ({
      type: '--generated create--',
      types: createActionTypes,
      url,
      requestOpts: { ...requestOpts,
        method: 'POST',
        body: JSON.stringify(body),
      },
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }

  // id is required as an argument to update a record;
  // we have a scenario with profiless, where it's
  // profile/{id}/image,
  // which is awkward...
  const updateActionTypes = filterActionTypes(actionTypes, 'update');
  if (updateActionTypes.length === 3) {
    actionCreators.update = ({
      id,
      body,
      onSuccess,
      onFailure,
    }) => ({
      type: '--generated update--',
      types: updateActionTypes,
      url: `${url}${id}/`,
      requestOpts: { ...requestOpts,
        method: 'PUT',
        body: JSON.stringify(body),
      },
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }


  return actionCreators;
};
