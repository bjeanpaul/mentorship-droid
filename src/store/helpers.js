import moment from 'moment';
import { values, isUndefined, sortBy, filter, first, orderBy } from 'lodash';

import { getCurrent } from 'src/navigationHelpers';
import { TYPES_TO_COLLECTIONS } from 'src/constants/events';


export const getAuthUserProfile = ({
  auth: { profileId },
  entities: { profiles },
}) => !isUndefined(profileId)
  ? profiles[profileId]
  : null;


export const getContext = state => {
  const { auth: { auth } } = state;

  return {
    auth,
    profile: getAuthUserProfile(state),
  };
};


export const getCategories = ({ entities }) =>
  sortBy(values(entities.categories), 'ordinal');


export const getCategoryActivities = ({ entities }, targetCategoryId) => {
  const activities = values(entities.activities)
    .filter(({ category }) => category === targetCategoryId);

  return sortBy(activities, 'ordinal');
};


export const getCategory = ({ entities: { categories } }, id) =>
  categories[id];


export const getActivity = ({ entities: { activities } }, id) =>
  activities[id];


export const getScheduledCalls = ({ entities: { scheduledCalls } }) =>
  values(scheduledCalls);


export const getScheduledCall = ({ entities: { scheduledCalls } }, id) =>
  scheduledCalls[id];


export const getScheduledCallActivity = (state, id) => {
  const scheduledCall = getScheduledCall(state, id);
  return scheduledCall
      && scheduledCall.activity
      && getActivity(state, scheduledCall.activity);
};


export const getActivityCallNotes = ({
  entities: { callNotes },
}, activityId) =>
  values(callNotes)
    .filter(({ callActivityId }) => callActivityId === activityId);


export const getCallNotes = ({
  entities: { callNotes },
}) =>
  sortBy(callNotes, ({ callStartTime }) => +moment(callStartTime));


export const getCall = ({ entities: { calls } }, id) => calls[id];


export const getCallNote = ({ entities: { callNotes } }, id) => callNotes[id];


export const getNextScheduledCall = (state, predicate = {}, time = Date.now()) => {
  let res = sortBy(getScheduledCalls(state), ({ callTime }) => +moment(callTime));
  res = res.filter(({ callTime }) => moment(callTime).isSameOrAfter(time));
  res = filter(res, predicate);
  return first(res);
};


export const getScheduledCallsBetween = (state, start, end, time = Date.now()) => {
  // for some reason, moment durations are behaving inconsistently in tests,
  // thus the lack of moment usage here
  const startTime = +moment(time) - start;
  const endTime = +moment(time) + end;

  return getScheduledCalls(state)
    .filter(({ callTime }) => {
      const t = +moment(callTime);
      return startTime <= t && t <= endTime;
    });
};


const getEventObject = (state, event) => {
  const collection = state.entities[TYPES_TO_COLLECTIONS[event.eventType]];
  return collection && collection[event.objectId];
};


// TODO return the associated entitites to make the containers' jobs easier
export const getEvents = (state) =>
  values(state.entities.events)
    .filter(event => getEventObject(state, event));


export const getMessages = ({ entities }) => []
  .concat(values(entities.messages))
  .concat(values(entities.pendingMessages));


export const getMessage = ({ entities }, id) => entities.messages[id];


export const getActiveTopRoute = store => getCurrent(store.navigation.top);


export const getBlogPost = ({ entities }, id) => entities.blogPosts[id];


export const getBlogPosts = ({ entities }) =>
  orderBy(entities.blogPosts, [({ createdAt }) => +moment(createdAt)], ['desc']);
