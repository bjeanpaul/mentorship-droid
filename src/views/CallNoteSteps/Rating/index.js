import React, { PropTypes } from 'react';
import { View, ScrollView } from 'react-native';
import { RadioList } from 'src/components';

import { FormStep } from 'src/components';
import styles from './styles';


const Rating = ({
  activityHelpful = '',
  onChange,
  ...props,
}) => {
  const items = [
    'Not at all',
    'A little',
    'Somewhat',
    'Quite a bit',
    'A lot',
  ];
  const initialSelectedIndex = items.indexOf(activityHelpful);

  return (
    <FormStep
      paginationDisabled={activityHelpful.length === 0}
      title="Was the activity helpful in facilitating your discussion?"
      {...props}
    >
      <View style={styles.ratingContainer}>
        <ScrollView>
          <View>
            <RadioList
              uid="ratings"
              items={items}
              onIndexChanged={(item) => onChange({ activityHelpful: item.item })}
              initialSelectedIndex={
                initialSelectedIndex < 0
                  ? initialSelectedIndex
                  : void 0
              }
            />
          </View>
        </ScrollView>
      </View>
    </FormStep>
  );
};

Rating.propTypes = {
  activityHelpful: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};


export default Rating;
