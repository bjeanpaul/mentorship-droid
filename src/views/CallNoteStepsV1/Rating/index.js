import React, { PropTypes } from 'react';
import { View, ScrollView } from 'react-native';
import { RadioList } from 'src/components';

import { FormStep } from 'src/components';
import styles from 'src/views/CallNoteStepsV1/styles';
import * as constants from 'src/constants/callNotes';


const Rating = ({
  onChange,
  callNote: { activityHelpful = -1 },
  ...props,
}) => {
  return (
    <FormStep
      paginationDisabled={activityHelpful.length === 0}
      title="Was the activity helpful in facilitating your discussion?"
      {...props}
    >
      <View style={styles.ratingContainer}>
        <ScrollView>
          <View>
            <RadioList
              uid="ratings"
              items={constants.RATING_ITEMS}
              onIndexChanged={item => onChange({
                activityHelpful: item.index.toString(),
              })}
              initialSelectedIndex={+activityHelpful}
            />
          </View>
        </ScrollView>
      </View>
    </FormStep>
  );
};

Rating.propTypes = {
  callNote: PropTypes.object,
  onChange: PropTypes.func.isRequired,
};


export default Rating;
