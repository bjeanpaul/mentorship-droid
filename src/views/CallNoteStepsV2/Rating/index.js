import { isUndefined, map } from 'lodash';
import React, { PropTypes } from 'react';
import { ScrollView } from 'react-native';

import * as constants from 'src/constants/callNotes';
import { FormStep, Radio, RadioItem } from 'src/components';


const Rating = ({
  callNote: { rating },
  onChange,
  ...props,
}) => (
  <FormStep
    paginationDisabled={isUndefined(rating)}
    title="How would you rate this activity?"
    {...props}
  >
    <ScrollView>
      <Radio
        uid="ratingItems"
        selection={rating}
        onSelect={selection => onChange({ rating: selection })}
      >
      {
        map(constants.V2_RATING_LABELS, (label, value) =>
          <RadioItem key={value} value={value}>{label}</RadioItem>)
      }
      </Radio>
    </ScrollView>
  </FormStep>
);

Rating.propTypes = {
  callNote: PropTypes.object,
  onChange: PropTypes.func.isRequired,
};


export default Rating;
