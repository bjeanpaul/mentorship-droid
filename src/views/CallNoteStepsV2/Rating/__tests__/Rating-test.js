import { noop } from 'lodash';
import React from 'react';

import { uidEquals, fakeCallNoteV2 } from 'app/scripts/helpers';
import Rating from 'src/views/CallNoteStepsV2/Rating';


describe('Rating', () => {
  const createComponent = (props = {}) => (
    <Rating
      callNote={fakeCallNoteV2()}
      onChange={noop}
      onBackPress={noop}
      onNextPress={noop}
      {...props}
    />
  );

  it('should render', () => {
    const el = render(createComponent());
    expect(el.toJSON()).toMatchSnapshot();
  });

  it('should call onChange() when the selection changes', () => {
    const onChange = jest.fn();
    const el = shallow(createComponent({ onChange }));

    el.findWhere(uidEquals('ratingItems'))
      .simulate('select', '3');

    expect(onChange.mock.calls)
      .toEqual([[{ rating: '3' }]]);
  });
});
