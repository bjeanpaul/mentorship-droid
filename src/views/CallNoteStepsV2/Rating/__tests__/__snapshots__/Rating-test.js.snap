exports[`Rating should render 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      undefined,
    ]
  }>
  <View
    style={
      Object {
        "margin": 25,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
          },
          Object {
            "fontSize": 13,
            "textAlign": "center",
          },
          Array [
            Object {
              "fontFamily": "Brandon Text_medium",
              "fontSize": 20,
              "fontWeight": "400",
            },
            Object {
              "margin": 13,
            },
          ],
        ]
      }>
      How would you rate this activity?
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }>
    <ScrollView>
      <View
        style={
          Object {
            "flex": 1,
            "marginBottom": 16,
            "marginLeft": 40,
            "marginRight": 40,
          }
        }>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#dfe5e6",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingBottom": 18,
              "paddingTop": 18,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#003035",
                  "fontFamily": "Brandon Text",
                },
                Object {
                  "fontSize": 13,
                  "textAlign": "center",
                },
                Object {
                  "color": "#003035",
                  "flex": 1,
                  "fontSize": 18,
                  "marginRight": 16,
                  "textAlign": "left",
                },
              ]
            }>
            Very good
          </Text>
          <Image
            source={1}
            style={
              Object {
                "height": 32,
                "width": 32,
              }
            } />
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#dfe5e6",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingBottom": 18,
              "paddingTop": 18,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#003035",
                  "fontFamily": "Brandon Text",
                },
                Object {
                  "fontSize": 13,
                  "textAlign": "center",
                },
                Object {
                  "color": "#003035",
                  "flex": 1,
                  "fontSize": 18,
                  "marginRight": 16,
                  "textAlign": "left",
                },
              ]
            }>
            Good
          </Text>
          <Image
            source={1}
            style={
              Object {
                "height": 32,
                "width": 32,
              }
            } />
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#dfe5e6",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingBottom": 18,
              "paddingTop": 18,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#003035",
                  "fontFamily": "Brandon Text",
                },
                Object {
                  "fontSize": 13,
                  "textAlign": "center",
                },
                Object {
                  "color": "#003035",
                  "flex": 1,
                  "fontSize": 18,
                  "marginRight": 16,
                  "textAlign": "left",
                },
              ]
            }>
            Average
          </Text>
          <Image
            source={1}
            style={
              Object {
                "height": 32,
                "width": 32,
              }
            } />
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#dfe5e6",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingBottom": 18,
              "paddingTop": 18,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#003035",
                  "fontFamily": "Brandon Text",
                },
                Object {
                  "fontSize": 13,
                  "textAlign": "center",
                },
                Object {
                  "color": "#003035",
                  "flex": 1,
                  "fontSize": 18,
                  "marginRight": 16,
                  "textAlign": "left",
                },
              ]
            }>
            Poor
          </Text>
          <Image
            source={1}
            style={
              Object {
                "height": 32,
                "width": 32,
              }
            } />
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#dfe5e6",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingBottom": 18,
              "paddingTop": 18,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#003035",
                  "fontFamily": "Brandon Text",
                },
                Object {
                  "fontSize": 13,
                  "textAlign": "center",
                },
                Object {
                  "color": "#003035",
                  "flex": 1,
                  "fontSize": 18,
                  "marginRight": 16,
                  "textAlign": "left",
                },
              ]
            }>
            Very Poor
          </Text>
          <Image
            source={1}
            style={
              Object {
                "height": 32,
                "width": 32,
              }
            } />
        </View>
      </View>
    </ScrollView>
  </View>
  <View
    style={
      Object {
        "justifyContent": "flex-end",
      }
    }>
    <View
      style={
        Object {
          "flexDirection": "row",
          "margin": 16,
        }
      }>
      <TouchableNativeFeedback
        onPress={[Function]}>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 5,
                "justifyContent": "center",
                "marginHorizontal": 16,
                "paddingBottom": 17,
                "paddingTop": 15,
              },
              Object {
                "backgroundColor": "transparent",
                "borderColor": "#f77040",
                "borderWidth": 1,
              },
              Array [
                Object {
                  "marginLeft": 0,
                  "marginRight": 0,
                },
                Object {
                  "marginRight": 16,
                  "width": 58,
                },
              ],
              undefined,
              undefined,
            ]
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#003035",
                  "fontFamily": "Brandon Text",
                },
                Object {
                  "fontSize": 13,
                  "textAlign": "center",
                },
                Array [
                  Object {
                    "fontFamily": "Brandon Text_bold",
                    "fontSize": 13,
                    "fontWeight": "bold",
                    "letterSpacing": 1,
                  },
                  Object {
                    "color": "#f77040",
                  },
                  undefined,
                  undefined,
                ],
              ]
            }>
            <Image
              source={1}
              style={
                Array [
                  Object {
                    "height": 24,
                    "width": 24,
                  },
                  Object {},
                ]
              } />
          </Text>
        </View>
      </TouchableNativeFeedback>
      <TouchableNativeFeedback
        onPress={[Function]}>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 5,
                "justifyContent": "center",
                "marginHorizontal": 16,
                "paddingBottom": 17,
                "paddingTop": 15,
              },
              Object {
                "backgroundColor": "transparent",
                "borderColor": "#f77040",
                "borderWidth": 1,
              },
              Array [
                Object {
                  "marginLeft": 0,
                  "marginRight": 0,
                },
                Object {
                  "flex": 1,
                },
              ],
              undefined,
              false,
            ]
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#003035",
                  "fontFamily": "Brandon Text",
                },
                Object {
                  "fontSize": 13,
                  "textAlign": "center",
                },
                Array [
                  Object {
                    "fontFamily": "Brandon Text_bold",
                    "fontSize": 13,
                    "fontWeight": "bold",
                    "letterSpacing": 1,
                  },
                  Object {
                    "color": "#f77040",
                  },
                  undefined,
                  false,
                ],
              ]
            }>
            NEXT
          </Text>
        </View>
      </TouchableNativeFeedback>
    </View>
  </View>
</View>
`;
